test/junit/src/test/scala/strawman/collection/IteratorTest.scala

```diff
@@ -315,4 +315,40 @@ class IteratorTest {
 
     assertEquals("a,null,b,null,c,null", it.mkString(","))
   }
+
+  @Test
+  def emptyTypedIteratorsShouldBeEqual: Unit = {
+    val emptyDoubleIterator = Iterator.empty[Double]
+    val emptyIntIterator = Iterator.empty[Int]
+    assertEquals(emptyDoubleIterator, emptyIntIterator)
```

Comment:
use assertSame when the results should be eq

Reply:
Fixed, thanks for pointing that out.